xdsl/interactive/app.py

```diff
@@ -91,6 +91,11 @@ class InputApp(App[None]):
         }
         """
 
+    ctx: MLContext
```

Comment:
I think it's quite important to not add mutable state to the app, could you please refactor this to avoid passing the context? Probably something like this:
```suggestion
    all_dialects: tuple[tuple[str, Callable[[], Dialect]]]
```
And just pass that in the init. Similarly, `all_passes` below should be passed in the init. This will also massively improve our tests, as we can initialize the test app with a smaller set of dialects and passes for it to condense.

Reply:
This seems like a good idea. I'm leaning towards making both "all_passes" and "all_dialects" into dictionaries. I don't really see why they should be tuples of tuples?

Reply:
because tuples are frozen, mostly. a frozendict would also work, but requires an extra dependency